### PR TITLE
feat: migrateTo

### PR DESCRIPTION
**Pull request by simonDos**
Add `migrateTo` function to PolygonMigration contract.

### CODE REVIEW

File: src/PolygonMigration.sol [hunk 1]
```diff
@@ -51,6 +51,14 @@ contract PolygonMigration is Ownable2StepUpgradeable, IPolygonMigration {
         polygon.safeTransfer(msg.sender, amount);
     }
 
+    /// @inheritdoc IPolygonMigration
+    function migrateTo(address recipient, uint256 amount) external {
+        emit Migrated(msg.sender, amount);
```
Comment (web3security) [inline src/PolygonMigration.sol]: Just wondering whether or not we should follow the same standard used by Unmigrated where we do have both the origin account and recipient as part of the event

    event Unmigrated(address indexed account, address indexed recipient, uint256 amount);


We should probably create a new event following the same pattern for MigrationTo in order to capture "msg.sender vs. recipient information" to differentiate from a normal migrate function.

wdyt?

File: src/PolygonMigration.sol [hunk 2]
```diff
@@ -51,6 +51,14 @@ contract PolygonMigration is Ownable2StepUpgradeable, IPolygonMigration {
...
+
+        matic.safeTransferFrom(msg.sender, address(this), amount);
+        polygon.safeTransfer(recipient, amount);
```
Comment (web3security) [inline src/PolygonMigration.sol:59]: I would suggest we require at least to have recipient != address(0) .

**Review by web3security — COMMENTED** [review]
My

File: src/PolygonMigration.sol (hunk 1 above)
Comment (simonDos) [reply src/PolygonMigration.sol]: agreed and added.

File: src/PolygonMigration.sol (hunk 2 above)
Comment (simonDos) [reply src/PolygonMigration.sol:59]: hm we don't have this in any of the functions so far. I think at this point we usually leave this layer of protection to the wallets.

Comment (web3security) [reply src/PolygonMigration.sol:59]: Hi, thanks. I'd suggest we confirm if this event signature "change" is breaking any known relevant event listeners to agree on the best way to mitigate its impact (i.e. sometimes new events were created).

File: src/interfaces/IPolygonMigration.sol [hunk 3]
```diff
@@ -11,8 +11,9 @@ import {IERC20} from "openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
 interface IPolygonMigration {
     /// @notice emitted when MATIC are migrated to POL
     /// @param account the account that migrated MATIC
+    /// @param recipient the account that received POL
     /// @param amount the amount of MATIC that was migrated
-    event Migrated(address indexed account, uint256 amount);
+    event Migrated(address indexed account, address recipient, uint256 amount);
```
Comment (web3security) [inline src/interfaces/IPolygonMigration.sol:16]: Hi, thanks. I'd suggest we confirm if this event signature "change" is breaking any known relevant event listeners to agree on the best way to mitigate its impact (i.e. sometimes we do recommend to create new events instead).

File: src/PolygonMigration.sol (hunk 2 above)
Comment (web3security) [reply src/PolygonMigration.sol:59]: acknowledged

**Review by web3security — APPROVED** [review]
Let's just check the event signature change likely breaking any relevant listener to decide whether or not it is better to add a new event for this "migrateTo". The rest LGTM. Thanks!

**Pull request closed without merging**